### PR TITLE
File upload

### PR DESCRIPTION
**Pull request by ericroberts**
- [x] Upload a file
- [x] Commit it to git

### CODE REVIEW

**Comment by ericroberts** [comment]
This closes #21

File: app.rb [hunk 1]
```diff
@@ -119,6 +119,30 @@ def project_path
...
+      path = "#{project_path}/#{params['splat'].join('/')}"
+      upload(path, params['file'])
+    end
+
+    def upload(path, file)
+      directory = Directory.new(path)
+      file_path = "#{path}/#{file[:filename]}"
+
+      ::File.open(file_path, "w") do |f|
+        f.write(file[:tempfile].read)
```
Comment (singpolyma) [inline app.rb]: should stream from tempfile to file.  or maybe we can even mv it?  Though it's probably possibly in RAM, so stream is the safer bet.

File: app.rb [hunk 2]
```diff
@@ -119,6 +119,30 @@ def project_path
...
+      end
+
+      g = Git.open(project_path)
+      g.add(file_path)
+      g.commit("Uploaded #{file[:filename]} via tint")
```
Comment (singpolyma) [inline app.rb]: This commit can fail, if we upload an overwrite that is identical to the current file.  Should check as in PUT

**Comment by singpolyma** [comment]
merged to master

**Pull request closed without merging**